commands/moderation/approve.js

```diff
@@ -0,0 +1,63 @@
+const { DevApplication } = require('../../src/models')
+
+exports.run = async (handler, message, args, pre) => {
+	let responseMessage
+	let member = message.guild.members.get(args.user)
+	let devApplication = await DevApplication.where().findOne({ userId: args.user }).exec()
+
+	if (devApplication) {
+		let log = message.guild.channels.find(c => c.name === 'mod-log')
+		let msg = await log.fetchMessage(devApplication.messageId)
+		let msgEmbed = msg.embeds[0]
+		// remove circular json
+		delete msgEmbed.author.embed
+		let embed = {
+			author: msgEmbed.author,
+			title: msgEmbed.title,
+			fields: msgEmbed.fields.map(field => {
+				// remove circular json
+				delete field.embed
+				return field
+			}),
+			color: msgEmbed.color
+		}
+
+		if (args.approved) {
+			let developerRole = message.guild.roles.find(r => r.name.toLowerCase() === 'developer')
+			let helperRole = message.guild.roles.find(r => r.name.toLowerCase() === 'helper')
+			await member.addRole(developerRole)
+			if (devApplication.helper) member.addRole(helperRole)
+			await member.send(`Your application for the Developer and/or Helper roles has been approved!\n${args.message}`)
+			responseMessage = 'Application approved, member informed.'
+			embed.color = 0x00ff00
+		} else {
+			await member.send(`Your application for the Developer and/or Helper roles has been denied.\n${args.message}`)
+			responseMessage = 'Application denied, member informed.'
+			embed.color = 0xff0000
+		}
+		await msg.edit({ embed: embed })
+		await DevApplication.where().findOneAndDelete({ userId: args.user }).exec()
```

Comment:
Is there a reason why this isn't just `await devApplication.remove()`?

(And if you wish to keep it as it is you need to query by `guildId` as well; currently this will delete all applications of this user from all guilds.)

Reply:
`remove` is deprecated, so using `deleteOne` instead.